packages/docusaurus-plugin-content-blog/src/__tests__/blogUtils.test.ts

```diff
@@ -272,3 +273,220 @@ describe('linkify', () => {
     } as BlogBrokenMarkdownLink);
   });
 });
+
+describe('blog sort', () => {
+  const pluginDir = 'blog-with-ref';
+
+  const descendingBlogPost: BlogPost[] = [
+    {
+      id: 'newest',
+      metadata: {
+        permalink: '/blog/2018/12/14/Happy-First-Birthday-Slash',
+        source: path.posix.join(
+          '@site',
+          pluginDir,
+          '2018-12-14-Happy-First-Birthday-Slash.md',
+        ),
+        title: 'Happy 1st Birthday Slash!',
+        description: `pattern name`,
+        date: new Date('2018-12-14'),
+        tags: [],
+        prevItem: {
+          permalink: '/blog/2019/01/01/date-matter',
+          title: 'date-matter',
+        },
+        hasTruncateMarker: false,
+        frontMatter: {},
+        authors: [],
+        unlisted: false,
+        formattedDate: '',
+      },
+      content: '',
```

Comment:
too verbose test code, create a helper that can accept partial blog post args and let us focus on what matters

Your test should be as readable as this (pseudo code), with the intent being super clear

```js


it("sorts descending by default",() => {
  const blogPost2022 = createBlogPost({date: new Date('2022-12-14')});
	const blogPost2023 = createBlogPost({date: new Date('2023-12-14')});
	const blogPost2024 = createBlogPost({date: new Date('2024-12-14')});

  assert(sort([
	  blogPost2022,
	  blogPost2024,
	  blogPost2023,
  ])).equals([
		blogPost2024,
		blogPost2023,
		blogPost2022
	])
});
```

It's preferable if each test create its own inputs, instead of relying on a shared list of posts for all tests. As we add tests, this list will only grow over time and tests will become harder to understand and maintain